zendesk/user_test.go

```diff
@@ -72,3 +72,20 @@ func TestCreateUser(t *testing.T) {
 		t.Fatal("Failed to create user")
 	}
 }
+
+func TestCreateUOrUpdateUser(t *testing.T) {
+	mockAPI := newMockAPIWithStatus(http.MethodPost, "users.json", http.StatusCreated)
```

Comment:
we should probably add an additional test to this method to make sure the client can handle 200s back from the server as well.

Reply:
So my xp on testing is low as we use a weird system at work and that is mostly my experience. Is there an example in this code base I can build off of?

Reply:
You could largely base it off what you have already written here!

so in this test here you specify that you are looking to receive a 201 response code from the server. That is what the `http.StatusCreated` bit is about. 

For this API though you need to be certain that the client can also handle 200s coming back from the server as that is how it will respond in the case of an update.

Basically i would suggest copying this test to another one and changing the status code you are looking for / verifying that the client doesn't return an error when that status code is returned. Hope that helps